extensions/gmsa-dc/v1/Promote-DC.ps1

```diff
@@ -0,0 +1,58 @@
+<#
+ .SYNOPSIS
+ This script will automate the creation of a Forest, Domain, and Domain Controller
+
+ .NOTES
+ This is only for automated e2e testing.  DO NOT use this for production.
+ Jeremy Wood (JeremyWx)
+ Version: 1.0.0.0
+#>
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+$PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
+
+# Create a working directory and change to it
+mkdir C:\gmsa
+Set-Location -Path C:\gmsa
+
+# Logging for troubleshooting
+Start-Transcript -Path "C:\gmsa\Promote.txt"
+
+# Download gMSA Setup Script
+Invoke-WebRequest -UseBasicParsing https://raw.githubusercontent.com/kubernetes-sigs/windows-testing/master/extensions/gmsa-dc/v1/Setup-gMSA.ps1 -OutFile Setup-gMSA.ps1
+
+#Install NuGet
+Install-PackageProvider -Name NuGet -Force
+
+#Create Local Administrator
+$admpassword = ( "K8s" + -join ((48..57) + (97..122) | Get-Random -Count 64 | ForEach-Object {[char]$_}) )
+$admpassword_secure = ( $admpassword | ConvertTo-SecureString -AsPlainText -Force)
+New-LocalUser -Name gmsa-admin -Password $admpassword_secure
+Add-LocalGroupMember -Group "Administrators" -Member gmsa-admin
+
+
+# Make Setup-gMSA run on next boot
+$Logon = "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon"
+$RunOnce = "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\RunOnce"
+Set-ItemProperty $Logon "AutoAdminLogon" -Value "1" -type String
+Set-ItemProperty $Logon "AutoLogonCount" -Value "2" -type DWord
+Set-ItemProperty $Logon "DefaultUsername" -Value "k8sgmsa\gmsa-admin" -type String
+Set-ItemProperty $Logon "DefaultPassword" -Value "$admpassword" -type String
+New-ItemProperty $RunOnce "gmsa" -Value "C:\Windows\system32\WindowsPowerShell\v1.0\powershell.exe -command C:\gmsa\Setup-gMSA.ps1" -Type String
+
+# Install and Enable SSH Server
+$SSHService = Get-WindowsCapability -Online | Where-Object Name -like "OpenSSH.Server*"
```

Comment:
Aks-engine already installs ssh server for Windows if it's enabled via template. Wondering if this will fail if there is an existing installation. If so, would be great to check for the service before installing or manage any resulting error. I'll check ASAP what happens if it already exists, haven't really taken it into account when I tested the extension.

Reply:
Nevermind. I tried it out, doesn't error out if it already exists.